_data-prepper/pipelines/configuration/processors/csv.md

```diff
@@ -23,9 +23,78 @@ delete_header | No | Boolean | If specified, the event header (`column_names_sou
 column_names_source_key | No | String | The field in the event that specifies the CSV column names, which will be automatically detected. If there need to be extra column names, the column names are automatically generated according to their index. If `column_names` is also defined, the header in `column_names_source_key` can also be used to generate the event fields. If too few columns are specified in this field, the remaining column names are automatically generated. If too many column names are specified in this field, the CSV processor omits the extra column names.
 column_names | No | List | User-specified names for the CSV columns. Default value is `[column1, column2, ..., columnN]` if there are no columns of data in the CSV record and `column_names_source_key` is not defined. If `column_names_source_key` is defined, the header in `column_names_source_key` generates the event fields. If too few columns are specified in this field, the remaining column names are automatically generated. If too many column names are specified in this field, the CSV processor omits the extra column names.
 
-<!---## Configuration
+## Usage
 
-Content will be added to this section.--->
+Add the following examples to your `pipelines.yaml` file, depending on how you your CSV columns are formatted.
+
+### User-specified column names
+
+The following example `pipelines.yaml` configuration points to a file named `ingest.csv` as the source. Then, the `csv` processor parses the data from the `.csv` file using the column names specified in the `column_names` setting, as shown in the following example:
+
+```yaml
+csv-pipeline:
+  source:
+    file:
+      path: "/full/path/to/ingest.csv"
+      record_type: "event"
+  processor:
+    - csv:
+        column_names: ["col1", "col2"]
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+
+When run, the processor will parse the message. Although only two column names are specified in processor settings, a third column name is autogenerated since the data contained in `ingest.csv` included three columns, `1,2,3`:
```

Comment:
```suggestion
When run, the processor will parse the message. Although only two column names are specified in processor settings, a third column name is automatically generated since the data contained in `ingest.csv` includes three columns, `1,2,3`:
```